File: packages/python-sdk/strait/operations/batch_operations.py
"""Batch operation management."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class BatchOperationsService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/batch-operations", query=query)

    def get(self, batch_id: str) -> dict[str, Any]:
        return self._request(
            "GET", "/v1/batch-operations/{batchID}", path_params={"batchID": batch_id},
        )


class AsyncBatchOperationsService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/batch-operations", query=query)

    async def get(self, batch_id: str) -> dict[str, Any]:
        return await self._request(
            "GET", "/v1/batch-operations/{batchID}", path_params={"batchID": batch_id},
        )

File: packages/ruby-sdk/lib/strait/operations/batch_operations.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for batch operation tracking.
    class BatchOperationsService < BaseService
      # List all batch operations.
      # GET /v1/batch-operations
      def list(query: nil)
        _request(:get, "/v1/batch-operations", query: query)
      end

      # Get a batch operation by ID.
      # GET /v1/batch-operations/{batchID}
      def get(batch_id)
        _request(:get, "/v1/batch-operations/{batchID}", path_params: { "batchID" => batch_id })
      end
    end
  end
end
